flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/crd/status/Savepoint.java

```diff
@@ -38,7 +36,42 @@ public class Savepoint {
     /** Savepoint trigger mechanism. */
     private SavepointTriggerType triggerType = SavepointTriggerType.UNKNOWN;
 
-    public static Savepoint of(String location, SavepointTriggerType type) {
-        return new Savepoint(System.currentTimeMillis(), location, type);
+    private SavepointFormatType formatType = SavepointFormatType.UNKNOWN;
+
+    /**
+     * Nonce value used when the savepoint was triggered manually {@link
+     * SavepointTriggerType#MANUAL}, defaults to 0.
+     */
+    private Long triggerNonce = 0L;
+
+    public Savepoint(
+            long timeStamp,
+            String location,
+            SavepointTriggerType triggerType,
+            SavepointFormatType formatType,
+            Long triggerNonce) {
+        this.timeStamp = timeStamp;
+        this.location = location;
+        this.triggerType = triggerType;
+        this.formatType = formatType;
+        setTriggerNonce(triggerNonce);
+    }
+
+    public static Savepoint of(String location, SavepointTriggerType triggerType) {
+        return new Savepoint(
+                System.currentTimeMillis(), location, triggerType, SavepointFormatType.UNKNOWN, 0L);
+    }
+
+    public static Savepoint of(
+            String location, SavepointTriggerType triggerType, SavepointFormatType formatType) {
+        return new Savepoint(System.currentTimeMillis(), location, triggerType, formatType, 0L);
+    }
```

Comment:
What is the purpose of these methods if you don't pass the savepointTriggerNonce ? You added logic to handle the nulls but you still explicitly pass `0L` here.

Reply:
I could've used nulls in the static helpers, but I added the null handling for a different purpose. Mainly to handle deserialization scenarios for older savepoints with no nonce. I guess it's using the default constructors + setters.

Reply:
The static helpers where there for convenince.